---
title: "scRNAseq_comparisons"
author: "<center> Author: Oliver Polzer <center><br>"
date: "<center> _`r Sys.Date()`_ <center>"
output: html_document

---

```{r setup, include=FALSE}
library(rmarkdown)
library(tinytex)
library(knitr)
knitr::opts_chunk$set(echo=TRUE, warning=FALSE, message=FALSE, cache = TRUE)
```

### 1. Background

The following analysis was carried out based on publicly available datasets  
1.[Shin et al., 2015](https://doi.org/10.1016/j.stem.2015.07.013)   
2.[Artegiani et al., 2017](https://doi.org/10.1016/j.celrep.2017.11.050)  
3.[Hochgerner er al., 2018](https://doi.org/10.1038/s41593-017-0056-2)  
4.[Batiuk et al., 2020](https://doi.org/10.1038/s41467-019-14198-8)  
5.[Harris et al., 2021](https://doi.org/10.1016/j.stem.2021.01.003)  
6.[Bielefeld et al., 2024 ](https://doi.org/10.1101/2023.03.04.531101)  
This reproducible and dynamic report was created using Rmarkdown and the [Knitr package](https://yihui.name/knitr/), and summarizes the basic code and outputs (plots, tables, etc).

***

Information regarding the separated studies can be found in Table 2 of the paper.

<br/>
Overview of the conducted analysis in this literature review:  

![](Workflow_scRNAseq_datasets.png){width="50%"}  

***


### 2. Figure 1

#### Gene lists used to identify/characterize neural stem cells or neural progenitor cells were extracted from the studies above and used for hierarchical clustering using the hclust function based on the similiarity of genes across studies. | Code

```{r clustering}

#Load necessary packages
library(dplyr)
library(VennDiagram)
library(gplots)
library(dendextend)
library(UpSetR)
library(plotly)
library(gridExtra)
library("ggvenn")

# For Figure 1 A
## Read and store genes----
# Set the path to the directory where your gene list files of are located
# All gene lists can be found in the folder GeneLists_NSC_NPC
gene_list_dir <- "C:/Users/opolzer/OneDrive - UvA/Desktop/Ph.D/Review_NSC_scRNAseq/R_code/GeneLists_NSC_NPC"

# List all files in the directory
gene_list_files <- list.files(gene_list_dir, pattern = ".txt", full.names = TRUE)

# Read and store gene lists in a list of data frames
gene_lists <- lapply(gene_list_files, function(file) {
  # Extract the file name without extension
  new_name <- tools::file_path_sans_ext(basename(file))
  
  # Read the gene list from the file
  gene_list <- read.delim(file, header = FALSE, stringsAsFactors = FALSE)
  
  # Rename the gene list with the new name
  names(gene_list) <- new_name
  
  # Return the gene list
  gene_list
})

# Perform pairwise comparisons and store the results----
pairwise_results <- list()

for (i in 1:(length(gene_lists) - 1)) {
  for (j in (i+1):length(gene_lists)) {
    set1 <- gene_lists[[i]][[1]]
    set2 <- gene_lists[[j]][[1]]
    common_genes <- intersect(set1, set2)
    pairwise_results[[paste0("Gene List", i, "_Gene List", j)]] <- common_genes
  }
}

# Quantification of Similarity
similarity_matrix <- matrix(0, nrow = length(gene_lists), ncol = length(gene_lists))

for (i in 1:(length(gene_lists) - 1)) {
  for (j in (i+1):length(gene_lists)) {
    set1 <- gene_lists[[i]][[1]]
    set2 <- gene_lists[[j]][[1]]
    common_genes <- intersect(set1, set2)
    similarity_matrix[i, j] <- length(common_genes)
    similarity_matrix[j, i] <- length(common_genes)
  }
}

# Print similarity matrix
rownames(similarity_matrix) <- c("Artegiani_NP", "Artegiani_NSC", 
                                 "Batiuk_AST4",
                                 "Bielefeld_NSC_stage1_2","Bielefeld_RGL",
                                 "Harris_NSC_IPCs","Harris_NSC",
                                 "Hochgerner_nIPC", "Hochgerner_RGL",
                                 "Shin_NSC")

colnames(similarity_matrix) <- c("Artegiani_NP", "Artegiani_NSC", 
                                 "Batiuk_AST4",
                                 "Bielefeld_NSC_stage1_2","Bielefeld_RGL",
                                 "Harris_NSC_IPCs","Harris_NSC",
                                 "Hochgerner_nIPC", "Hochgerner_RGL",
                                 "Shin_NSC")



# Create a dendrogram using hierarchical clustering
dendrogram <- hclust(as.dist(1 - (similarity_matrix / max(similarity_matrix))))



# Add color to the dendrogram branches based on clusters
clusters <- cutree(dendrogram, k = 2)  

# Define color palette for clusters
cluster_colors <- c("red", "black")

# Assign colors to dendrogram branches
dendrogram_colored <- color_branches(dendrogram, k = 2, col = cluster_colors)

# Create a color palette for the heatmap
heatmap_colors <- colorRampPalette(c("white", "blue"))(100)


#For Figure 1 B
## Read and store genes----
# Set the path to the directory where your gene list files are located
# All gene lists can be found in the folder GeneLists_Astro
gene_list_dir2 <- "C:/Users/opolzer/OneDrive - UvA/Desktop/Ph.D/Review_NSC_scRNAseq/R_code/GeneLists_Astro"

# List all files in the directory
gene_list_files2 <- list.files(gene_list_dir2, pattern = ".txt", full.names = TRUE)

# Read and store gene lists in a list of data frames
gene_lists2 <- lapply(gene_list_files2, function(file) {
  # Extract the file name without extension
  new_name <- tools::file_path_sans_ext(basename(file))
  
  # Read the gene list from the file
  gene_list <- read.delim(file, header = FALSE, stringsAsFactors = FALSE)
  
  # Rename the gene list with the new name
  names(gene_list) <- new_name
  
  # Return the gene list
  gene_list
})

# Perform pairwise comparisons and store the results----
pairwise_results2 <- list()

for (i in 1:(length(gene_lists2) - 1)) {
  for (j in (i+1):length(gene_lists2)) {
    set1 <- gene_lists2[[i]][[1]]
    set2 <- gene_lists2[[j]][[1]]
    common_genes2 <- intersect(set1, set2)
    pairwise_results[[paste0("Gene List", i, "_Gene List", j)]] <- common_genes2
  }
}

# Quantification of Similarity
similarity_matrix2 <- matrix(0, nrow = length(gene_lists2), ncol = length(gene_lists2))

for (i in 1:(length(gene_lists2) - 1)) {
  for (j in (i+1):length(gene_lists2)) {
    set1 <- gene_lists2[[i]][[1]]
    set2 <- gene_lists2[[j]][[1]]
    common_genes2 <- intersect(set1, set2)
    similarity_matrix2[i, j] <- length(common_genes2)
    similarity_matrix2[j, i] <- length(common_genes2)
  }
}

# Print similarity matrix
# Print similarity matrix
rownames(similarity_matrix2) <- c("Bielefeld_A_stage1","Bielefeld_A_stage2","Bielefeld_A_stage3",  "Bielefeld_A_stage4","Bielefeld_A_stage5","Bielefeld_A_stage6","Bielefeld_A_stage7","Bielefeld_NSC_stage1","Bielefeld_NSC_stage2","Bielefeld_RGL",
                                 "Karpf_Cluster0", "Karpf_Cluster1", "Karpf_Cluster2", "Karpf_Cluster3", "Karpf_Cluster4")

colnames(similarity_matrix2) <- c("Bielefeld_A_stage1","Bielefeld_A_stage2","Bielefeld_A_stage3",  "Bielefeld_A_stage4","Bielefeld_A_stage5","Bielefeld_A_stage6","Bielefeld_A_stage7","Bielefeld_NSC_stage1","Bielefeld_NSC_stage2","Bielefeld_RGL",
                                 "Karpf_Cluster0", "Karpf_Cluster1", "Karpf_Cluster2", "Karpf_Cluster3", "Karpf_Cluster4")



# Create a dendrogram using hierarchical clustering
dendrogram2 <- hclust(as.dist(1 - (similarity_matrix2 / max(similarity_matrix2))))



# Add color to the dendrogram branches based on clusters
clusters2 <- cutree(dendrogram2, k = 5)  

# Define color palette for clusters
cluster_colors2 <- c("red", "black", "green", "blue", "orange")

# Assign colors to dendrogram branches
dendrogram_colored2 <- color_branches(dendrogram2, k = 5, col = cluster_colors2)

# Create a color palette for the heatmap
heatmap_colors2 <- colorRampPalette(c("white", "blue"))(100)

```

#### List of gene numbers shared between individual studies and the visualization as a heatmap 

```{r Results similarity matrix, list, echo = FALSE}

print(similarity_matrix)

```


</br>

#### Hierarchical clustering across the different NSC and NP populations | Figure 1 A

</br>

```{r Results dendrograms NSC NP, echo = FALSE}

# Plot the dendrogram
plot(dendrogram, main = "Dendrogram", xlab = "Gene Lists", ylab = "Distance")


# Plot the colored dendrogram
plot(dendrogram_colored, main = "Colored Dendrogram", ylab = "Distance")

```

***

</br>

#### Hierarchical clustering across the different astrocytic populations | Figure 1 B

</br>

```{r Results dendrograms Astro, echo = FALSE}

# Plot the dendrogram
plot(dendrogram2, main = "Dendrogram", xlab = "Gene Lists", ylab = "Distance")


# Plot the colored dendrogram
plot(dendrogram_colored2, main = "Colored Dendrogram", ylab = "Distance")

```

***

###  Upset plots of shared genes of the different NSC and NP populations | Code for Figure 1 C and D 

```{r Upset plots}

#Load packages----
library(UpSetR)
library(ComplexHeatmap)
library(ggplot2)
library(ggpubr)
library(svglite)
library(ggplotify)

#Load gene lists of NSC markers----
Shin_NSC <- readLines("Shin_NSC.txt")
Artegiani_NSC <- readLines("Artegiani_NSC.txt")
Hochgerner_RGL <- readLines("Hochgerner_RGL.txt")
Harris_NSC <- readLines("Harris_NSC_pos.txt")

Artegiani_NP  <- readLines("Artegiani_NP.txt")
Hochgerner_nIPC <- readLines("Hochgerner_nIPC.txt")
Batiuk_AST4 <- readLines("Batiuk_AST4.txt")
Harris_NSC_IPCs <- readLines("Harris_NSC_IPCs_pos.txt")
Bielefeld_RGL <- readLines("Bielefeld_RGL.txt")
Bielefeld_NSC_stage1_2 <- readLines("Bielefeld_NSC_stage1_2.txt")


#Generate lists of sets (with and without Bielefeld et al.) + binary matrix

list1 <- list( Shin_NSC,
               Artegiani_NSC,
               Hochgerner_RGL,
               Harris_NSC
               )
matrix1 <- list_to_matrix(list1)
colnames(matrix1) <- c("Shin et al, 2015","Artegiani et al., 2017","Batiuk et al., 2020","Harris et al., 2021")

list2 <- list( Artegiani_NP,
               Hochgerner_nIPC,
               Batiuk_AST4,
               Harris_NSC_IPCs,
               Bielefeld_RGL,
               Bielefeld_NSC_stage1_2
               )
matrix2 <- list_to_matrix(list2)
colnames(matrix2) <- c("Artegiani et al, 2017","Hochgerner et al., 2018","Batiuk et al., 2020","Harris et al., 2021", "Bielefeld et al., 2024 (RGL)","Bielefeld et al., 2024 (NSC)")
#Generate the Upset plot
m1 = make_comb_mat(matrix1)
graph1 <- UpSet(t(m1[comb_degree(m1) >= 2]), 
      top_annotation = upset_top_annotation(t(m1[comb_degree(m1) >= 2]), add_numbers = TRUE),
      right_annotation = upset_right_annotation(t(m1[comb_degree(m1) >= 2]), add_numbers = TRUE),
      set_order = c("Shin et al, 2015","Artegiani et al., 2017","Batiuk et al., 2020","Harris et al., 2021"))

m2 = make_comb_mat(matrix2)
graph2 <- UpSet(t(m2[comb_degree(m2) >= 4]), 
                top_annotation = upset_top_annotation(t(m2[comb_degree(m2) >= 4]), add_numbers = TRUE),
                right_annotation = upset_right_annotation(t(m2[comb_degree(m2) >= 4]), add_numbers = TRUE),
                set_order = c("Artegiani et al, 2017","Hochgerner et al., 2018","Batiuk et al., 2020","Harris et al., 2021","Bielefeld et al., 2024 (RGL)","Bielefeld et al., 2024 (NSC)"))


```

</br>

#### Upset plots of shared gene signatures across the different studies | Figure 1 C and 1 D

```{r Upset plots results, echo=FALSE }
graph1_ggplot <- as.ggplot(graph1)
graph2_ggplot <- as.ggplot(graph2)
ggarrange(graph1_ggplot,graph2_ggplot)
```

### 3. Figure 2

Pseudotime inference aims to infer the ordering of cells along a lineage based on the cells’ gene expression profiles measured by scRNA-seq, and the inferential target is “pseudotime,” a time-like variable indicating the relative position a cell takes in a lineage. By establishing a temporal dimension in a static scRNA-seq dataset, pseudotime inference allows the probing of individual genes’ expression dynamics along with continuous cell-state changes. If a gene’s mean expression changes along pseudotime, the gene is referred to as differentially expressed (DE) and is likely to play an important role in the underlying cellular process that gives rise to the pseudotime.

The first study reporting a methhod for a developmental trajectory in adult hippocampal neural stem cells was [Shin et al., 2015](https://www.cell.com/cell-stem-cell/fulltext/S1934-5909(15)00312-4?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1934590915003124%3Fshowall%3Dtrue) using a newly developed pipeline called Waterfall. 

A second study with the introduction of resting, dormant, and active adult hippocampal neural stem cells was [Harris et al., 2021](https://www.cell.com/cell-stem-cell/fulltext/S1934-5909(21)00003-5?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS1934590921000035%3Fshowall%3Dtrue#supplementaryMaterial). Here they used he trajectory tool Slingshot to infer progression along a pseudotime curve from quiescence to activation on their final dataset of NSCs. This method was also validated with the Shin et al. paper and shows substantial overlap of pseudotime genes. 

#### Code

```{r Pseudotime code}
library(readxl)
library(dplyr)
library(ggplot2)
library(tidyverse)
library(ggpubr)
library(ggdist)
library(ggforce)
library(xtable)
library(ggthemes)

######## Figure 2 A - D-----
# Specify the path to your Excel file of both pseudotime trajectories, can be found under the folder Pseudotime
 
excel_file_shin <- "C:/Users/opolzer/OneDrive - UvA/Desktop/Ph.D/Review_NSC_scRNAseq/R_code/Pseudotime/Shin_Pseudotime_all_genes.xlsx"
excel_file_harris <- "C:/Users/opolzer/OneDrive - UvA/Desktop/Ph.D/Review_NSC_scRNAseq/R_code/Pseudotime/Harris_Pseudotime_all_genes.csv"

# Read the Excel file into a data frame
data_shin <- read_excel(excel_file_shin)
data_shin_df <- data.frame(data_shin) %>%
  rename(GeneName =1)

data_shin_df2 <- data_shin_df[,-1]
rownames(data_shin_df2) <- data_shin_df[,1]

data_harris <- read.csv(excel_file_harris)
data_harris_df <- data.frame(data_harris) %>%
  rename(GeneName =1)

data_harris_df2 <- data_harris_df[,-1]
rownames(data_harris_df2) <- data_harris_df[,1]


# all gene lists of NSC populations from selected studies ----
Shin_NSC <- readLines("Shin_NSC.txt")
Artegiani_NSC <- readLines("Artegiani_NSC.txt")
Hochgerner_RGL <- readLines("Hochgerner_RGL.txt")
Harris_NSC <- readLines("Harris_NSC_pos.txt")
Overlap_NSC <- readLines("Overlap_NSC.txt")


# Filter the data frame based on selected genes for Shin and Harris ----

filtered_data_s_Shin_NSC <- data_shin_df2[rownames(data_shin_df2) %in% Shin_NSC, ]
filtered_data_h_Shin_NSC <- data_harris_df2[rownames(data_harris_df2) %in% Shin_NSC, ] %>%   
                          mutate_if(is.character, as.numeric)

filtered_data_s_Artegiani_NSC <- data_shin_df2[rownames(data_shin_df2) %in% Artegiani_NSC, ]
filtered_data_h_Artegiani_NSC <- data_harris_df2[rownames(data_harris_df2) %in% Artegiani_NSC, ] %>%   
                          mutate_if(is.character, as.numeric)

filtered_data_s_Hochgerner_RGL <- data_shin_df2[rownames(data_shin_df2) %in% Hochgerner_RGL, ]
filtered_data_h_Hochgerner_RGL <- data_harris_df2[rownames(data_harris_df2) %in% Hochgerner_RGL, ] %>%
                          mutate_if(is.character, as.numeric)

filtered_data_s_Harris_NSC <- data_shin_df2[rownames(data_shin_df2) %in% Harris_NSC, ]
filtered_data_h_Harris_NSC <- data_harris_df2[rownames(data_harris_df2) %in% Harris_NSC, ] %>%   
                          mutate_if(is.character, as.numeric)

filtered_data_s_Overlap_NSC <- data_shin_df2[rownames(data_shin_df2) %in% Overlap_NSC, ]
filtered_data_h_Overlap_NSC <- data_harris_df2[rownames(data_harris_df2) %in% Overlap_NSC, ] %>%   
                          mutate_if(is.character, as.numeric)


# Calculate the mean of each column in the filtered data for Shin and Harris 

mean_values_s_Shin_NSC <- colMeans(filtered_data_s_Shin_NSC)
mean_values_h_Shin_NSC <- colMeans(filtered_data_h_Shin_NSC)

mean_values_s_Artegiani_NSC <- colMeans(filtered_data_s_Artegiani_NSC)
mean_values_h_Artegiani_NSC <- colMeans(filtered_data_h_Artegiani_NSC)

mean_values_s_Hochgerner_RGL <- colMeans(filtered_data_s_Hochgerner_RGL)
mean_values_h_Hochgerner_RGL <- colMeans(filtered_data_h_Hochgerner_RGL)

mean_values_s_Harris_NSC <- colMeans(filtered_data_s_Harris_NSC)
mean_values_h_Harris_NSC <- colMeans(filtered_data_h_Harris_NSC)

mean_values_s_Overlap_NSC <- colMeans(filtered_data_s_Overlap_NSC)
mean_values_h_Overlap_NSC <- colMeans(filtered_data_h_Overlap_NSC)

# Convert the first row to a data frame
x_values_s <- as.data.frame(t(data_shin_df2[1, ]))
x_values_h <- as.data.frame(t(data_harris_df2[2, ]))%>%   
              mutate_if(is.character, as.numeric)

# Combine the x and y values into a single data frame

plot_data_s_Shin_NSC <- cbind(x_values_s, mean_values_s_Shin_NSC)
plot_data_h_Shin_NSC <- cbind(x_values_h, mean_values_h_Shin_NSC)

plot_data_s_Artegiani_NSC <- cbind(x_values_s, mean_values_s_Artegiani_NSC)
plot_data_h_Artegiani_NSC <- cbind(x_values_h, mean_values_h_Artegiani_NSC)

plot_data_s_Hochgerner_RGL <- cbind(x_values_s, mean_values_s_Hochgerner_RGL)
plot_data_h_Hochgerner_RGL <- cbind(x_values_h, mean_values_h_Hochgerner_RGL)

plot_data_s_Harris_NSC <- cbind(x_values_s, mean_values_s_Harris_NSC)
plot_data_h_Harris_NSC <- cbind(x_values_h, mean_values_h_Harris_NSC)

plot_data_s_Overlap_NSC <- cbind(x_values_s, mean_values_s_Overlap_NSC)
plot_data_h_Overlap_NSC <- cbind(x_values_h, mean_values_h_Overlap_NSC)

# Z-Score Normalization
plot_data_s_Shin_NSC$mean_values_s_zscore <- scale(plot_data_s_Shin_NSC$mean_values_s_Shin_NSC)
plot_data_h_Shin_NSC$mean_values_h_zscore <- scale(plot_data_h_Shin_NSC$mean_values_h_Shin_NSC)

plot_data_s_Artegiani_NSC$mean_values_s_zscore <- scale(plot_data_s_Artegiani_NSC$mean_values_s_Artegiani_NSC)
plot_data_h_Artegiani_NSC$mean_values_h_zscore <- scale(plot_data_h_Artegiani_NSC$mean_values_h_Artegiani_NSC)

plot_data_s_Hochgerner_RGL$mean_values_s_zscore <- scale(plot_data_s_Hochgerner_RGL$mean_values_s_Hochgerner_RGL)
plot_data_h_Hochgerner_RGL$mean_values_h_zscore <- scale(plot_data_h_Hochgerner_RGL$mean_values_h_Hochgerner_RGL)

plot_data_s_Harris_NSC$mean_values_s_zscore <- scale(plot_data_s_Harris_NSC$mean_values_s_Harris_NSC)
plot_data_h_Harris_NSC$mean_values_h_zscore <- scale(plot_data_h_Harris_NSC$mean_values_h_Harris_NSC)

plot_data_s_Overlap_NSC$mean_values_s_zscore <- scale(plot_data_s_Overlap_NSC$mean_values_s_Overlap_NSC)
plot_data_h_Overlap_NSC$mean_values_h_zscore <- scale(plot_data_h_Overlap_NSC$mean_values_h_Overlap_NSC)



windowsFonts("Arial" = windowsFont("Arial"))

# Combined datasets on one plot NSC Figure 1 A and B----
# Combine datasets for s_ prefix (Shin pseudotime)----
# Rename columns to ensure consistency
plot_data_s_Overlap_NSC <- plot_data_s_Overlap_NSC[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Shin_NSC <- plot_data_s_Shin_NSC[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Artegiani_NSC <- plot_data_s_Artegiani_NSC[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Hochgerner_RGL <- plot_data_s_Hochgerner_RGL[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Harris_NSC <- plot_data_s_Harris_NSC[, c("Pseudotime", "mean_values_s_zscore")]

# Add dataset column
plot_data_s_Overlap_NSC$dataset <- "s_Overlap_NSC"
plot_data_s_Shin_NSC$dataset <- "s_Shin_NSC"
plot_data_s_Artegiani_NSC$dataset <- "s_Artegiani_NSC"
plot_data_s_Hochgerner_RGL$dataset <- "s_Hochgerner_RGL"
plot_data_s_Harris_NSC$dataset <- "s_Harris_NSC"

plot_data_s_combined <- rbind(
  transform(plot_data_s_Overlap_NSC, dataset = "s_Overlap_NSC"),
  transform(plot_data_s_Shin_NSC, dataset = "s_Shin_NSC"),
  transform(plot_data_s_Artegiani_NSC, dataset = "s_Artegiani_NSC"),
  transform(plot_data_s_Hochgerner_RGL, dataset = "s_Hochgerner_RGL"),
  transform(plot_data_s_Harris_NSC, dataset = "s_Harris_NSC")
)


# Plot for NSC datasets on Shin pseudotime
s_combined_plot <- ggplot(plot_data_s_combined) +
  geom_point(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), size = 1,alpha = 0.2, show.legend = TRUE) +
  geom_smooth(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), method = "loess", se = FALSE) +
  labs(
    x = "Pseudotime",
    y = "Mean gene expression levels"
  ) +
  theme(
    plot.margin = margin(1, 1, 1, 1, "cm"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title.x = element_text(size = 18, face = "bold"),
    axis.title.y = element_text(size = 18, face = "bold"),
    axis.text = element_text(size = 18),
    legend.position = "bottom",
    legend.text = element_text(size = 16)
  ) +
  scale_color_manual(values = c(
    "s_Overlap_NSC" = "#6c9c96ff", 
    "s_Shin_NSC" = "#9c4445ff", 
     "s_Artegiani_NSC" = "#4f7c7cff", 
     "s_Hochgerner_RGL" = "#c54d4dff", 
     "s_Harris_NSC" = "#8a9a5bff"), labels = c("s_Overlap_NSC" = "Shared NSC genes Fig.1", 
    "s_Shin_NSC" = "NSC cluster (Shin et al.)", 
     "s_Artegiani_NSC" = "NSC cluster (Artegiani et al.)", 
     "s_Hochgerner_RGL" = "RGL cluster (Hochgerner et al.)", 
     "s_Harris_NSC" = "NSC cluster (Harris et al.)"), name = NULL
  ) +
  xlim(0, 1) +
  ylim(-2, 2) +
  theme_tufte(base_family = "Arial") +
  theme(aspect.ratio = 0.8)  # Adjust aspect ratio to make it narrower

print(s_combined_plot)

# Combine datasets for h_ prefix (Harris pseudotime)----
# Standardize columns for h_ data frames
plot_data_h_Overlap_NSC <- plot_data_h_Overlap_NSC[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Shin_NSC <- plot_data_h_Shin_NSC[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Artegiani_NSC <- plot_data_h_Artegiani_NSC[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Hochgerner_RGL <- plot_data_h_Hochgerner_RGL[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Harris_NSC <- plot_data_h_Harris_NSC[, c("Lineage1", "mean_values_h_zscore")]

# Add dataset column
plot_data_h_Overlap_NSC$dataset <- "h_Overlap_NSC"
plot_data_h_Shin_NSC$dataset <- "h_Shin_NSC"
plot_data_h_Artegiani_NSC$dataset <- "h_Artegiani_NSC"
plot_data_h_Hochgerner_RGL$dataset <- "h_Hochgerner_RGL"
plot_data_h_Harris_NSC$dataset <- "h_Harris_NSC"

# Combine datasets for h_ prefix
plot_data_h_combined <- rbind(
  transform(plot_data_h_Overlap_NSC, dataset = "h_Overlap_NSC"),
  transform(plot_data_h_Shin_NSC, dataset = "h_Shin_NSC"),
  transform(plot_data_h_Artegiani_NSC, dataset = "h_Artegiani_NSC"),
  transform(plot_data_h_Hochgerner_RGL, dataset = "h_Hochgerner_RGL"),
  transform(plot_data_h_Harris_NSC, dataset = "h_Harris_NSC")
)


# Plot for h_ datasets
h_combined_plot <- ggplot(plot_data_h_combined) +
  geom_point(aes(x = Lineage1, y = mean_values_h_zscore, color = dataset), size = 1, alpha = 0.2, show.legend = TRUE) +
  geom_smooth(aes(x = Lineage1, y = mean_values_h_zscore, color = dataset), method = "loess", se = FALSE, size=1.5) +
  labs(
    x = "Pseudotime",
    y = "Mean gene expression level"
  ) +
  theme(
    plot.margin = margin(1, 1, 1, 1, "cm"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title.x = element_text(size = 16),
    axis.title.y = element_text(size = 16),
    axis.text = element_text(size = 16),
  ) +
  scale_color_manual(values = c(
    "h_Overlap_NSC" = "#6c9c96ff", 
    "h_Shin_NSC" = "#9c4445ff", 
     "h_Artegiani_NSC" = "#4f7c7cff", 
     "h_Hochgerner_RGL" = "#c54d4dff", 
     "h_Harris_NSC" = "#8a9a5bff"), labels = c("h_Overlap_NSC" = "Shared NSC genes Fig.1", 
    "h_Shin_NSC" = "NSC cluster (Shin et al.)", 
     "h_Artegiani_NSC" = "NSC cluster (Artegiani et al.)", 
     "h_Hochgerner_RGL" = "RGL cluster (Hochgerner et al.)", 
     "h_Harris_NSC" = "NSC cluster (Harris et al.)"), name = NULL
  )  +
  xlim(0, 20) +
  ylim(-2, 5)
print(h_combined_plot)


# all gene lists of NP populations from selected studies----
Hochgerner_nIPC <- readLines("Hochgerner_nIPC.txt")
Batiuk_AST4 <- readLines("Batiuk_AST4.txt")
Artegiani_NP <- readLines("Artegiani_NP.txt")
Harris_NSC_IPC <- readLines("Harris_NSC_IPCs_pos.txt")
Bielefeld_NSC_1_2 <- readLines("Bielefeld_NSC_stage1_2.txt")
Bielefeld_RGL <- readLines("Bielefeld_RGL.txt")
Overlap_NP <- readLines("Overlap_NP.txt")

# Filter the data frame based on selected genes for Shin and Harris ----
filtered_data_h_Hochgerner_nIPC <- data_harris_df2[rownames(data_harris_df2) %in% Hochgerner_nIPC, ]%>%   
                          mutate_if(is.character, as.numeric)
filtered_data_s_Hochgerner_nIPC <- data_shin_df2[rownames(data_shin_df2) %in% Hochgerner_nIPC, ]

filtered_data_h_Batiuk_AST4 <- data_harris_df2[rownames(data_harris_df2) %in% Batiuk_AST4, ]%>%   
                          mutate_if(is.character, as.numeric)
filtered_data_s_Batiuk_AST4 <- data_shin_df2[rownames(data_shin_df2) %in% Batiuk_AST4, ]

filtered_data_h_Artegiani_NP <- data_harris_df2[rownames(data_harris_df2) %in% Artegiani_NP, ]%>%   
                          mutate_if(is.character, as.numeric)
filtered_data_s_Artegiani_NP <- data_shin_df2[rownames(data_shin_df2) %in% Artegiani_NP, ]

filtered_data_h_Harris_NSC_IPC <- data_harris_df2[rownames(data_harris_df2) %in% Harris_NSC_IPC, ]%>%   
                          mutate_if(is.character, as.numeric)
filtered_data_s_Harris_NSC_IPC <- data_shin_df2[rownames(data_shin_df2) %in% Harris_NSC_IPC, ]

filtered_data_h_Bielefeld_NSC_1_2 <- data_harris_df2[rownames(data_harris_df2) %in% Bielefeld_NSC_1_2, ] %>%
  mutate_if(is.character, as.numeric)
filtered_data_s_Bielefeld_NSC_1_2 <- data_shin_df2[rownames(data_shin_df2) %in% Bielefeld_NSC_1_2, ]

filtered_data_h_Bielefeld_RGL <- data_harris_df2[rownames(data_harris_df2) %in% Bielefeld_RGL, ] %>%
  mutate_if(is.character, as.numeric)
filtered_data_s_Bielefeld_RGL <- data_shin_df2[rownames(data_shin_df2) %in% Bielefeld_RGL, ]


filtered_data_h_Overlap_NP <- data_harris_df2[rownames(data_harris_df2) %in% Overlap_NP, ]%>%   
                          mutate_if(is.character, as.numeric)
filtered_data_s_Overlap_NP <- data_shin_df2[rownames(data_shin_df2) %in% Overlap_NP, ]



# Calculate the mean of each column in the filtered data
mean_values_s_Hochgerner_nIPC <- colMeans(filtered_data_s_Hochgerner_nIPC)
mean_values_h_Hochgerner_nIPC <- colMeans(filtered_data_h_Hochgerner_nIPC)

mean_values_s_Batiuk_AST4 <- colMeans(filtered_data_s_Batiuk_AST4)
mean_values_h_Batiuk_AST4 <- colMeans(filtered_data_h_Batiuk_AST4)

mean_values_s_Artegiani_NP <- colMeans(filtered_data_s_Artegiani_NP)
mean_values_h_Artegiani_NP <- colMeans(filtered_data_h_Artegiani_NP)

mean_values_s_Harris_NSC_IPC <- colMeans(filtered_data_s_Harris_NSC_IPC)
mean_values_h_Harris_NSC_IPC <- colMeans(filtered_data_h_Harris_NSC_IPC)

mean_values_s_Bielefeld_NSC_1_2 <- colMeans(filtered_data_s_Bielefeld_NSC_1_2)
mean_values_h_Bielefeld_NSC_1_2 <- colMeans(filtered_data_h_Bielefeld_NSC_1_2)

mean_values_s_Bielefeld_RGL <- colMeans(filtered_data_s_Bielefeld_RGL)
mean_values_h_Bielefeld_RGL <- colMeans(filtered_data_h_Bielefeld_RGL)

mean_values_s_Overlap_NP <- colMeans(filtered_data_s_Overlap_NP)
mean_values_h_Overlap_NP <- colMeans(filtered_data_h_Overlap_NP)

# Convert the first row to a data frame for x values
x_values_s <- as.data.frame(t(data_shin_df2[1, ]))
x_values_h <- as.data.frame(t(data_harris_df2[2, ])) %>%
              mutate_if(is.character, as.numeric)

# Combine the x and y values into a single data frame
plot_data_s_Hochgerner_nIPC <- cbind(x_values_s, mean_values_s_Hochgerner_nIPC)
plot_data_h_Hochgerner_nIPC <- cbind(x_values_h, mean_values_h_Hochgerner_nIPC)

plot_data_s_Batiuk_AST4 <- cbind(x_values_s, mean_values_s_Batiuk_AST4)
plot_data_h_Batiuk_AST4 <- cbind(x_values_h, mean_values_h_Batiuk_AST4)

plot_data_s_Artegiani_NP <- cbind(x_values_s, mean_values_s_Artegiani_NP)
plot_data_h_Artegiani_NP <- cbind(x_values_h, mean_values_h_Artegiani_NP)

plot_data_s_Harris_NSC_IPC <- cbind(x_values_s, mean_values_s_Harris_NSC_IPC)
plot_data_h_Harris_NSC_IPC <- cbind(x_values_h, mean_values_h_Harris_NSC_IPC)

plot_data_s_Bielefeld_NSC_1_2 <- cbind(x_values_s, mean_values_s_Bielefeld_NSC_1_2)
plot_data_h_Bielefeld_NSC_1_2 <- cbind(x_values_h, mean_values_h_Bielefeld_NSC_1_2)

plot_data_s_Bielefeld_RGL <- cbind(x_values_s, mean_values_s_Bielefeld_RGL)
plot_data_h_Bielefeld_RGL <- cbind(x_values_h, mean_values_h_Bielefeld_RGL)

plot_data_s_Overlap_NP <- cbind(x_values_s, mean_values_s_Overlap_NP)
plot_data_h_Overlap_NP <- cbind(x_values_h, mean_values_h_Overlap_NP)


# Z-Score Normalization
plot_data_s_Hochgerner_nIPC$mean_values_s_zscore <- scale(plot_data_s_Hochgerner_nIPC$mean_values_s_Hochgerner_nIPC)
plot_data_h_Hochgerner_nIPC$mean_values_h_zscore <- scale(plot_data_h_Hochgerner_nIPC$mean_values_h_Hochgerner_nIPC)

plot_data_s_Batiuk_AST4$mean_values_s_zscore <- scale(plot_data_s_Batiuk_AST4$mean_values_s_Batiuk_AST4)
plot_data_h_Batiuk_AST4$mean_values_h_zscore <- scale(plot_data_h_Batiuk_AST4$mean_values_h_Batiuk_AST4)

plot_data_s_Artegiani_NP$mean_values_s_zscore <- scale(plot_data_s_Artegiani_NP$mean_values_s_Artegiani_NP)
plot_data_h_Artegiani_NP$mean_values_h_zscore <- scale(plot_data_h_Artegiani_NP$mean_values_h_Artegiani_NP)

plot_data_s_Harris_NSC_IPC$mean_values_s_zscore <- scale(plot_data_s_Harris_NSC_IPC$mean_values_s_Harris_NSC_IPC)
plot_data_h_Harris_NSC_IPC$mean_values_h_zscore <- scale(plot_data_h_Harris_NSC_IPC$mean_values_h_Harris_NSC_IPC)

plot_data_s_Bielefeld_NSC_1_2$mean_values_s_zscore <- scale(plot_data_s_Bielefeld_NSC_1_2$mean_values_s_Bielefeld_NSC_1_2)
plot_data_h_Bielefeld_NSC_1_2$mean_values_h_zscore <- scale(plot_data_h_Bielefeld_NSC_1_2$mean_values_h_Bielefeld_NSC_1_2)

plot_data_s_Bielefeld_RGL$mean_values_s_zscore <- scale(plot_data_s_Bielefeld_RGL$mean_values_s_Bielefeld_RGL)
plot_data_h_Bielefeld_RGL$mean_values_h_zscore <- scale(plot_data_h_Bielefeld_RGL$mean_values_h_Bielefeld_RGL)

plot_data_s_Overlap_NP$mean_values_s_zscore <- scale(plot_data_s_Overlap_NP$mean_values_s_Overlap_NP)
plot_data_h_Overlap_NP$mean_values_h_zscore <- scale(plot_data_h_Overlap_NP$mean_values_h_Overlap_NP)

#Combined datasets on one plot----
# Standardize columns for new datasets
plot_data_s_Hochgerner_nIPC <- plot_data_s_Hochgerner_nIPC[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Batiuk_AST4 <- plot_data_s_Batiuk_AST4[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Artegiani_NP <- plot_data_s_Artegiani_NP[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Harris_NSC_IPC <- plot_data_s_Harris_NSC_IPC[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Bielefeld_NSC_1_2 <- plot_data_s_Bielefeld_NSC_1_2[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Bielefeld_RGL <- plot_data_s_Bielefeld_RGL[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Overlap_NP <- plot_data_s_Overlap_NP[, c("Pseudotime", "mean_values_s_zscore")]

# Add dataset column for s_ datasets
plot_data_s_Hochgerner_nIPC$dataset <- "s_Hochgerner_nIPC"
plot_data_s_Batiuk_AST4$dataset <- "s_Batiuk_AST4"
plot_data_s_Artegiani_NP$dataset <- "s_Artegiani_NP"
plot_data_s_Harris_NSC_IPC$dataset <- "s_Harris_NSC_IPC"
plot_data_s_Bielefeld_NSC_1_2$dataset <- "s_Bielefeld_NSC_1_2"
plot_data_s_Bielefeld_RGL$dataset <- "s_Bielefeld_RGL"
plot_data_s_Overlap_NP$dataset <- "s_Overlap_NP"

# Combine s_ datasets
plot_data_s_combined_new <- rbind(
  transform(plot_data_s_Hochgerner_nIPC, dataset = "s_Hochgerner_nIPC"),
  transform(plot_data_s_Batiuk_AST4, dataset = "s_Batiuk_AST4"),
  transform(plot_data_s_Artegiani_NP, dataset = "s_Artegiani_NP"),
  transform(plot_data_s_Harris_NSC_IPC, dataset = "s_Harris_NSC_IPC"),
   transform(plot_data_s_Bielefeld_NSC_1_2, dataset = "s_Bielefeld_NSC_1_2"),
  transform(plot_data_s_Bielefeld_RGL, dataset = "s_Bielefeld_RGL"),
  transform(plot_data_s_Overlap_NP, dataset = "s_Overlap_NP")
)

# Standardize columns for h_ datasets
plot_data_h_Hochgerner_nIPC <- plot_data_h_Hochgerner_nIPC[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Batiuk_AST4 <- plot_data_h_Batiuk_AST4[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Artegiani_NP <- plot_data_h_Artegiani_NP[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Harris_NSC_IPC <- plot_data_h_Harris_NSC_IPC[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Bielefeld_NSC_1_2 <- plot_data_h_Bielefeld_NSC_1_2[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Bielefeld_RGL <- plot_data_h_Bielefeld_RGL[, c("Lineage1", "mean_values_h_zscore")]
plot_data_h_Overlap_NP <- plot_data_h_Overlap_NP[, c("Lineage1", "mean_values_h_zscore")]

# Add dataset column for h_ datasets
plot_data_h_Hochgerner_nIPC$dataset <- "h_Hochgerner_nIPC"
plot_data_h_Batiuk_AST4$dataset <- "h_Batiuk_AST4"
plot_data_h_Artegiani_NP$dataset <- "h_Artegiani_NP"
plot_data_h_Harris_NSC_IPC$dataset <- "h_Harris_NSC_IPC"
plot_data_h_Bielefeld_NSC_1_2$dataset <- "h_Bielefeld_NSC_1_2"
plot_data_h_Bielefeld_RGL$dataset <- "h_Bielefeld_RGL"
plot_data_h_Overlap_NP$dataset <- "h_Overlap_NP"

# Combine h_ datasets
plot_data_h_combined_new <- rbind(
  transform(plot_data_h_Hochgerner_nIPC, dataset = "h_Hochgerner_nIPC"),
  transform(plot_data_h_Batiuk_AST4, dataset = "h_Batiuk_AST4"),
  transform(plot_data_h_Artegiani_NP, dataset = "h_Artegiani_NP"),
  transform(plot_data_h_Harris_NSC_IPC, dataset = "h_Harris_NSC_IPC"),
  transform(plot_data_h_Bielefeld_NSC_1_2, dataset = "h_Bielefeld_NSC_1_2"),
  transform(plot_data_h_Bielefeld_RGL, dataset = "h_Bielefeld_RGL"),
  transform(plot_data_h_Overlap_NP, dataset = "h_Overlap_NP")
)

# Plot for new s_ datasets
s_combined_plot_new <- ggplot(plot_data_s_combined_new) +
  geom_point(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), size = 1, alpha = 0.2, show.legend = TRUE) +
  geom_smooth(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), method = "loess", se = FALSE) +
  labs(
    x = "Pseudotime",
    y = "Mean gene expression level",
    title = "Combined datasets on Shin pseudotime (new gene sets)"
  ) +
  theme(
    plot.margin = margin(1, 1, 1, 1, "cm"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title.x = element_text(size = 14, face = "bold"),
    axis.title.y = element_text(size = 14, face = "bold"),
    axis.text = element_text(size = 14),
    legend.position = "bottom"
  ) +
  scale_color_manual(values = c(
    "s_Hochgerner_nIPC" = "#7f8c8d", 
    "s_Batiuk_AST4" = "#e67e22", 
    "s_Artegiani_NP" = "#3498db", 
    "s_Harris_NSC_IPC" = "#f39c12",
    "s_Bielefeld_NSC_1_2" = "#9b59b6",
    "s_Bielefeld_RGL" = "#1abc9c",
    "s_Overlap_NP" = "#2ecc71"), 
    labels = c("s_Hochgerner_nIPC" = "nIPC cluster (Hochgerner et al.)", 
    "s_Batiuk_AST4" = "AST4 cluster (Batiuk et al.)", 
    "s_Artegiani_NP" = "NP cluster (Artegiani et al.)", 
    "s_Harris_NSC_IPC" = "NSC-IPC cluster (Harris et al.)", 
     "s_Bielefeld_NSC_1_2" = "NSC 1/2 (Bielefeld et al.)",  
    "s_Bielefeld_RGL" = "RGL (Bielefeld et al.)" ,
    "s_Overlap_NP" = "Shared NP gene from Fig.1"), name = NULL
  ) +
  xlim(0, 1) +
  ylim(-2, 2) +
  theme_tufte(base_family = "Arial") +
  theme(aspect.ratio = 0.8)  # Adjust aspect ratio to make it narrower
print(s_combined_plot_new)

# Plot for new h_ datasets
h_combined_plot_new <- ggplot(plot_data_h_combined_new) +
  geom_point(aes(x = Lineage1, y = mean_values_h_zscore, color = dataset), size = 1,alpha = 0.2, show.legend = TRUE) +
  geom_smooth(aes(x = Lineage1, y = mean_values_h_zscore, color = dataset), method = "loess", se = FALSE, size=1.5) +
  labs(
    x = "Pseudotime",
    y = "Mean gene expression level"
  ) +
  theme(
    plot.margin = margin(1, 1, 1, 1, "cm"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title.x = element_text(size = 14, face = "bold"),
    axis.title.y = element_text(size = 14, face = "bold"),
    axis.text = element_text(size = 14)
  ) +
  scale_color_manual(values = c(
    "h_Hochgerner_nIPC" = "#7f8c8d", 
    "h_Batiuk_AST4" = "#e67e22", 
    "h_Artegiani_NP" = "#3498db", 
    "h_Harris_NSC_IPC" = "#f39c12",
    "h_Bielefeld_NSC_1_2" = "#9b59b6",
    "h_Bielefeld_RGL" = "#1abc9c",
    "h_Overlap_NP" = "#2ecc71"), 
    labels = c("h_Hochgerner_nIPC" = "nIPC cluster (Hochgerner et al.)", 
    "h_Batiuk_AST4" = "AST4 cluster (Batiuk et al.)", 
    "h_Artegiani_NP" = "NP cluster (Artegiani et al.)", 
    "h_Harris_NSC_IPC" = "NSC-IPC cluster (Harris et al.)", 
     "h_Bielefeld_NSC_1_2" = "NSC 1/2 (Bielefeld et al.)",  
    "h_Bielefeld_RGL" = "RGL (Bielefeld et al.)" ,
    "h_Overlap_NP" = "Shared NP gene from Fig.1"), name = NULL
  ) +
  xlim(0, 20) +
  ylim(-2, 5)

print(h_combined_plot_new)

######## Figure 2 E -F ----
# All gene lists from selected studies for all genes/top10 genes/overlap ----
TOP10_Batiuk <- readLines("TOP10_Batiuk.txt")
TOP10_Artegiani <- readLines("TOP10_Artegiani.txt")
TOP40_Artegiani <- readLines("TOP40_Artegiani.txt")
TOP40_Batiuk <- readLines("TOP40_Batiuk.txt")
TOP10_NSC_Harris <- readLines("TOP10_NSC_Harris.txt")
TOP40_NSC_Harris <- readLines("TOP40_NSC_Harris.txt")
TOP10_NSCIPC_Harris <- readLines("TOP10_NSCIPC_Harris.txt")
TOP40_NSCIPC_Harris <- readLines("TOP40_NSCIPC_Harris.txt")

# Filter data for each gene list ----
filtered_data_s_TOP10_Batiuk <- data_shin_df2[rownames(data_shin_df2) %in% TOP10_Batiuk, ]
filtered_data_h_TOP10_Batiuk <- data_harris_df2[rownames(data_harris_df2) %in% TOP10_Batiuk, ] %>%   
                                mutate_if(is.character, as.numeric)

filtered_data_s_TOP10_Artegiani <- data_shin_df2[rownames(data_shin_df2) %in% TOP10_Artegiani, ]
filtered_data_h_TOP10_Artegiani <- data_harris_df2[rownames(data_harris_df2) %in% TOP10_Artegiani, ] %>%   
                                   mutate_if(is.character, as.numeric)

filtered_data_s_TOP40_Artegiani <- data_shin_df2[rownames(data_shin_df2) %in% TOP40_Artegiani, ]
filtered_data_h_TOP40_Artegiani <- data_harris_df2[rownames(data_harris_df2) %in% TOP40_Artegiani, ] %>%   
                                   mutate_if(is.character, as.numeric)

filtered_data_s_TOP40_Batiuk <- data_shin_df2[rownames(data_shin_df2) %in% TOP40_Batiuk, ]
filtered_data_h_TOP40_Batiuk <- data_harris_df2[rownames(data_harris_df2) %in% TOP40_Batiuk, ] %>%   
                                mutate_if(is.character, as.numeric)

filtered_data_s_TOP10_NSC_Harris <- data_shin_df2[rownames(data_shin_df2) %in% TOP10_NSC_Harris, ]
filtered_data_h_TOP10_NSC_Harris <- data_harris_df2[rownames(data_harris_df2) %in% TOP10_NSC_Harris, ] %>%   
                                    mutate_if(is.character, as.numeric)

filtered_data_s_TOP40_NSC_Harris <- data_shin_df2[rownames(data_shin_df2) %in% TOP40_NSC_Harris, ]
filtered_data_h_TOP40_NSC_Harris <- data_harris_df2[rownames(data_harris_df2) %in% TOP40_NSC_Harris, ] %>%   
                                    mutate_if(is.character, as.numeric)

filtered_data_s_TOP10_NSCIPC_Harris <- data_shin_df2[rownames(data_shin_df2) %in% TOP10_NSCIPC_Harris, ]
filtered_data_h_TOP10_NSCIPC_Harris <- data_harris_df2[rownames(data_harris_df2) %in% TOP10_NSCIPC_Harris, ] %>%   
                                       mutate_if(is.character, as.numeric)

filtered_data_s_TOP40_NSCIPC_Harris <- data_shin_df2[rownames(data_shin_df2) %in% TOP40_NSCIPC_Harris, ]
filtered_data_h_TOP40_NSCIPC_Harris <- data_harris_df2[rownames(data_harris_df2) %in% TOP40_NSCIPC_Harris, ] %>%   
                                       mutate_if(is.character, as.numeric)

# Calculate mean values ----
mean_values_s_TOP10_Batiuk <- colMeans(filtered_data_s_TOP10_Batiuk)
mean_values_h_TOP10_Batiuk <- colMeans(filtered_data_h_TOP10_Batiuk)

mean_values_s_TOP10_Artegiani <- colMeans(filtered_data_s_TOP10_Artegiani)
mean_values_h_TOP10_Artegiani <- colMeans(filtered_data_h_TOP10_Artegiani)

mean_values_s_TOP40_Artegiani <- colMeans(filtered_data_s_TOP40_Artegiani)
mean_values_h_TOP40_Artegiani <- colMeans(filtered_data_h_TOP40_Artegiani)

mean_values_s_TOP40_Batiuk <- colMeans(filtered_data_s_TOP40_Batiuk)
mean_values_h_TOP40_Batiuk <- colMeans(filtered_data_h_TOP40_Batiuk)

mean_values_s_TOP10_NSC_Harris <- colMeans(filtered_data_s_TOP10_NSC_Harris)
mean_values_h_TOP10_NSC_Harris <- colMeans(filtered_data_h_TOP10_NSC_Harris)

mean_values_s_TOP40_NSC_Harris <- colMeans(filtered_data_s_TOP40_NSC_Harris)
mean_values_h_TOP40_NSC_Harris <- colMeans(filtered_data_h_TOP40_NSC_Harris)

mean_values_s_TOP10_NSCIPC_Harris <- colMeans(filtered_data_s_TOP10_NSCIPC_Harris)
mean_values_h_TOP10_NSCIPC_Harris <- colMeans(filtered_data_h_TOP10_NSCIPC_Harris)

mean_values_s_TOP40_NSCIPC_Harris <- colMeans(filtered_data_s_TOP40_NSCIPC_Harris)
mean_values_h_TOP40_NSCIPC_Harris <- colMeans(filtered_data_h_TOP40_NSCIPC_Harris)

# Combine data for plotting ----
plot_data_s_TOP10_Batiuk <- cbind(x_values_s, mean_values_s_TOP10_Batiuk)
plot_data_h_TOP10_Batiuk <- cbind(x_values_h, mean_values_h_TOP10_Batiuk)

plot_data_s_TOP10_Artegiani <- cbind(x_values_s, mean_values_s_TOP10_Artegiani)
plot_data_h_TOP10_Artegiani <- cbind(x_values_h, mean_values_h_TOP10_Artegiani)

plot_data_s_TOP40_Artegiani <- cbind(x_values_s, mean_values_s_TOP40_Artegiani)
plot_data_h_TOP40_Artegiani <- cbind(x_values_h, mean_values_h_TOP40_Artegiani)

plot_data_s_TOP40_Batiuk <- cbind(x_values_s, mean_values_s_TOP40_Batiuk)
plot_data_h_TOP40_Batiuk <- cbind(x_values_h, mean_values_h_TOP40_Batiuk)

plot_data_s_TOP10_NSC_Harris <- cbind(x_values_s, mean_values_s_TOP10_NSC_Harris)
plot_data_h_TOP10_NSC_Harris <- cbind(x_values_h, mean_values_h_TOP10_NSC_Harris)

plot_data_s_TOP40_NSC_Harris <- cbind(x_values_s, mean_values_s_TOP40_NSC_Harris)
plot_data_h_TOP40_NSC_Harris <- cbind(x_values_h, mean_values_h_TOP40_NSC_Harris)

plot_data_s_TOP10_NSCIPC_Harris <- cbind(x_values_s, mean_values_s_TOP10_NSCIPC_Harris)
plot_data_h_TOP10_NSCIPC_Harris <- cbind(x_values_h, mean_values_h_TOP10_NSCIPC_Harris)

plot_data_s_TOP40_NSCIPC_Harris <- cbind(x_values_s, mean_values_s_TOP40_NSCIPC_Harris)
plot_data_h_TOP40_NSCIPC_Harris <- cbind(x_values_h, mean_values_h_TOP40_NSCIPC_Harris)

# Z-Score Normalization ----
plot_data_s_TOP10_Batiuk$mean_values_s_zscore <- scale(plot_data_s_TOP10_Batiuk$mean_values_s_TOP10_Batiuk)
plot_data_h_TOP10_Batiuk$mean_values_h_zscore <- scale(plot_data_h_TOP10_Batiuk$mean_values_h_TOP10_Batiuk)

plot_data_s_TOP10_Artegiani$mean_values_s_zscore <- scale(plot_data_s_TOP10_Artegiani$mean_values_s_TOP10_Artegiani)
plot_data_h_TOP10_Artegiani$mean_values_h_zscore <- scale(plot_data_h_TOP10_Artegiani$mean_values_h_TOP10_Artegiani)

plot_data_s_TOP40_Artegiani$mean_values_s_zscore <- scale(plot_data_s_TOP40_Artegiani$mean_values_s_TOP40_Artegiani)
plot_data_h_TOP40_Artegiani$mean_values_h_zscore <- scale(plot_data_h_TOP40_Artegiani$mean_values_h_TOP40_Artegiani)

plot_data_s_TOP40_Batiuk$mean_values_s_zscore <- scale(plot_data_s_TOP40_Batiuk$mean_values_s_TOP40_Batiuk)
plot_data_h_TOP40_Batiuk$mean_values_h_zscore <- scale(plot_data_h_TOP40_Batiuk$mean_values_h_TOP40_Batiuk)

plot_data_s_TOP10_NSC_Harris$mean_values_s_zscore <- scale(plot_data_s_TOP10_NSC_Harris$mean_values_s_TOP10_NSC_Harris)
plot_data_h_TOP10_NSC_Harris$mean_values_h_zscore <- scale(plot_data_h_TOP10_NSC_Harris$mean_values_h_TOP10_NSC_Harris)

plot_data_s_TOP40_NSC_Harris$mean_values_s_zscore <- scale(plot_data_s_TOP40_NSC_Harris$mean_values_s_TOP40_NSC_Harris)
plot_data_h_TOP40_NSC_Harris$mean_values_h_zscore <- scale(plot_data_h_TOP40_NSC_Harris$mean_values_h_TOP40_NSC_Harris)

plot_data_s_TOP10_NSCIPC_Harris$mean_values_s_zscore <- scale(plot_data_s_TOP10_NSCIPC_Harris$mean_values_s_TOP10_NSCIPC_Harris)
plot_data_h_TOP10_NSCIPC_Harris$mean_values_h_zscore <- scale(plot_data_h_TOP10_NSCIPC_Harris$mean_values_h_TOP10_NSCIPC_Harris)

plot_data_s_TOP40_NSCIPC_Harris$mean_values_s_zscore <- scale(plot_data_s_TOP40_NSCIPC_Harris$mean_values_s_TOP40_NSCIPC_Harris)
plot_data_h_TOP40_NSCIPC_Harris$mean_values_h_zscore <- scale(plot_data_h_TOP40_NSCIPC_Harris$mean_values_h_TOP40_NSCIPC_Harris)



# Batiuk on Shin pseudotime ----
# Ensure consistency in column names
plot_data_s_TOP10_Batiuk <- plot_data_s_TOP10_Batiuk[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_TOP40_Batiuk <- plot_data_s_TOP40_Batiuk[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Overlap_NP <- plot_data_s_Overlap_NP[, c("Pseudotime", "mean_values_s_zscore")]

# Add dataset column
plot_data_s_TOP10_Batiuk$dataset <- "s_Top10_Batiuk"
plot_data_s_TOP40_Batiuk$dataset <- "s_Top40_Batiuk"
plot_data_s_Overlap_NP$dataset <- "s_Overlap_NP"

plot_data_s_combined_Batiuk <- rbind(
  transform(plot_data_s_Batiuk_AST4, dataset = "s_Batiuk_AST4"),
  transform(plot_data_s_TOP10_Batiuk, dataset = "s_Top10_Batiuk"),
  transform(plot_data_s_TOP40_Batiuk, dataset = "s_Top40_Batiuk"),
  transform(plot_data_s_Overlap_NP, dataset = "s_Overlap_NP")
)

# Plot for Batiuk on Shin pseudotime 
s_combined_plot_Batiuk <- ggplot(plot_data_s_combined_Batiuk) +
  geom_point(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), size = 1, alpha = 0.2, show.legend = TRUE) +
  geom_smooth(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), method = "loess", se = FALSE, size=1.5) +
  labs(
    x = "Pseudotime",
    y = "Mean gene expression levels",
    title = "AST4 cluster (Batiuk et al.)"
  ) +
  theme(
    plot.margin = margin(1, 1, 1, 1, "cm"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title.x = element_text(size = 18),
    axis.title.y = element_text(size = 18),
    axis.text = element_text(size = 18),
  ) +
  scale_color_manual(values = c(
    "s_Batiuk_AST4" = "#c54d4dff",
    "s_Top10_Batiuk" = "#7c4b9cff", 
    "s_Top40_Batiuk" = "#6b705cff", 
    "s_Overlap_NP" = "#ffad60ff"
  ), labels = c(
    "s_Batiuk_AST4" = "Whole gene set",
    "s_Top10_Batiuk" = "Top 10 genes", 
    "s_Top40_Batiuk" = "Top 40 genes", 
    "s_Overlap_NP" = "Shared NP genes Fig.1"
  ), name = NULL) +
  xlim(0, 1) +
  ylim(-2, 2) +
  theme_tufte(base_family = "Arial") +
  theme(aspect.ratio = 0.8)  # Adjust aspect ratio to make it narrower

# Display plot
print(s_combined_plot_Batiuk)


# Artegiani on Shin pseudotime ----

# Ensure consistency in column names
plot_data_s_TOP10_Artegiani <- plot_data_s_TOP10_Artegiani[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_TOP40_Artegiani <- plot_data_s_TOP40_Artegiani[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Overlap_NP <- plot_data_s_Overlap_NP[, c("Pseudotime", "mean_values_s_zscore")]

# Add dataset column
plot_data_s_TOP10_Artegiani$dataset <- "s_Top10_Artegiani"
plot_data_s_TOP40_Artegiani$dataset <- "s_Top40_Artegiani"
plot_data_s_Overlap_NP$dataset <- "s_Overlap_NP"

plot_data_s_combined_Artegiani <- rbind(
  transform(plot_data_s_Artegiani_NP, dataset = "s_Artegiani_NP"),
  transform(plot_data_s_TOP10_Artegiani, dataset = "s_Top10_Artegiani"),
  transform(plot_data_s_TOP40_Artegiani, dataset = "s_Top40_Artegiani"),
  transform(plot_data_s_Overlap_NP, dataset = "s_Overlap_NP")
)

# Plot for Artegiani on Shin pseudotime 
s_combined_plot_Artegiani <- ggplot(plot_data_s_combined_Artegiani) +
  geom_point(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), size = 1, alpha = 0.2, show.legend = TRUE) +
  geom_smooth(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), method = "loess", se = FALSE, size=1.5) +
  labs(
    x = "Pseudotime",
    y = "Mean gene expression levels",
    title = "NP cluster (Artegiani et al.)"
  ) +
  theme(
    plot.margin = margin(1, 1, 1, 1, "cm"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title.x = element_text(size = 18),
    axis.title.y = element_text(size = 18),
    axis.text = element_text(size = 18),
  ) +
  scale_color_manual(values = c(
    "s_Artegiani_NP" = "#c54d4dff",
    "s_Top10_Artegiani" = "#7c4b9cff", 
    "s_Top40_Artegiani" = "#6b705cff", 
    "s_Overlap_NP" = "#ffad60ff"
  ), labels = c(
    "s_Artegiani_NP" = "Whole gene set",
    "s_Top10_Artegiani" = "Top 10 genes", 
    "s_Top40_Artegiani" = "Top 40 genes", 
    "s_Overlap_NP" = "Shared NP genes Fig.1"
  ), name = NULL) +
  xlim(0, 1) +
  ylim(-2, 2) +
  theme_tufte(base_family = "Arial") +
  theme(aspect.ratio = 0.8)  # Adjust aspect ratio to make it narrower

# Display plot
print(s_combined_plot_Artegiani)


# NSC_Harris on Shin pseudotime ----

# Ensure consistency in column names
plot_data_s_TOP10_NSC_Harris <- plot_data_s_TOP10_NSC_Harris[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_TOP40_NSC_Harris <- plot_data_s_TOP40_NSC_Harris[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Overlap_NSC <- plot_data_s_Overlap_NSC[, c("Pseudotime", "mean_values_s_zscore")]

# Add dataset column
plot_data_s_TOP10_NSC_Harris$dataset <- "s_Top10_NSC_Harris"
plot_data_s_TOP40_NSC_Harris$dataset <- "s_Top40_NSC_Harris"
plot_data_s_Overlap_NSC$dataset <- "s_Overlap_NSC"

plot_data_s_combined_NSC_Harris <- rbind(
  transform(plot_data_s_Harris_NSC, dataset = "s_Harris_NSC"),
  transform(plot_data_s_TOP10_NSC_Harris, dataset = "s_Top10_NSC_Harris"),
  transform(plot_data_s_TOP40_NSC_Harris, dataset = "s_Top40_NSC_Harris"),
  transform(plot_data_s_Overlap_NSC, dataset = "s_Overlap_NSC")
)

# Plot for NSC_Harris on Shin pseudotime 
s_combined_plot_NSC_Harris <- ggplot(plot_data_s_combined_NSC_Harris) +
  geom_point(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), size = 1, alpha = 0.2, show.legend = TRUE) +
  geom_smooth(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), method = "loess", se = FALSE, size=1.5) +
  labs(
    x = "Pseudotime",
    y = "Mean gene expression levels",
    title = "NSC cluster (Harris et al.)"
  ) +
  theme(
    plot.margin = margin(1, 1, 1, 1, "cm"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title.x = element_text(size = 18),
    axis.title.y = element_text(size = 18),
    axis.text = element_text(size = 18),
  ) +
  scale_color_manual(values = c(
    "s_Harris_NSC" = "#c54d4dff",
    "s_Top10_NSC_Harris" = "#7c4b9cff", 
    "s_Top40_NSC_Harris" = "#6b705cff", 
    "s_Overlap_NSC" = "#ffad60ff"
  ), labels = c(
    "s_Harris_NSC" = "Whole gene set",
    "s_Top10_NSC_Harris" = "Top 10 genes", 
    "s_Top40_NSC_Harris" = "Top 40 genes", 
    "s_Overlap_NSC" = "Shared NSC genes Fig.1"
  ), name = NULL) +
  xlim(0, 1) +
  ylim(-2, 2) +
  theme_tufte(base_family = "Arial") +
  theme(aspect.ratio = 0.8)  # Adjust aspect ratio to make it narrower

# Display plot
print(s_combined_plot_NSC_Harris)



# NSCIPC_Harris on Shin pseudotime ----

# Ensure consistency in column names
plot_data_s_TOP10_NSCIPC_Harris <- plot_data_s_TOP10_NSCIPC_Harris[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_TOP40_NSCIPC_Harris <- plot_data_s_TOP40_NSCIPC_Harris[, c("Pseudotime", "mean_values_s_zscore")]
plot_data_s_Overlap_NP <- plot_data_s_Overlap_NP[, c("Pseudotime", "mean_values_s_zscore")]

# Add dataset column
plot_data_s_TOP10_NSCIPC_Harris$dataset <- "s_Top10_NSCIPC_Harris"
plot_data_s_TOP40_NSCIPC_Harris$dataset <- "s_Top40_NSCIPC_Harris"
plot_data_s_Overlap_NP$dataset <- "s_Overlap_NP"

plot_data_s_combined_NSCIPC_Harris <- rbind(
  transform(plot_data_s_Harris_NSC_IPC, dataset = "s_Harris_NSC_IPC"),
  transform(plot_data_s_TOP10_NSCIPC_Harris, dataset = "s_Top10_NSCIPC_Harris"),
  transform(plot_data_s_TOP40_NSCIPC_Harris, dataset = "s_Top40_NSCIPC_Harris"),
  transform(plot_data_s_Overlap_NP, dataset = "s_Overlap_NP")
)

# Plot for NSCIPC_Harris on Shin pseudotime 
s_combined_plot_NSCIPC_Harris <- ggplot(plot_data_s_combined_NSCIPC_Harris) +
  geom_point(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), size = 1, alpha = 0.2, show.legend = TRUE) +
  geom_smooth(aes(x = Pseudotime, y = mean_values_s_zscore, color = dataset), method = "loess", se = FALSE, size=1.5) +
  labs(
    x = "Pseudotime",
    y = "Mean gene expression levels",
    title = "NSC/IPC cluster (Harris et al.)"
  ) +
  theme(
    plot.margin = margin(1, 1, 1, 1, "cm"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.background = element_blank(),
    axis.line = element_line(colour = "black"),
    axis.title.x = element_text(size = 18),
    axis.title.y = element_text(size = 18),
    axis.text = element_text(size = 18),
  ) +
  scale_color_manual(values = c(
    "s_Harris_NSC_IPC" = "#c54d4dff",
    "s_Top10_NSCIPC_Harris" = "#7c4b9cff", 
    "s_Top40_NSCIPC_Harris" = "#6b705cff", 
    "s_Overlap_NP" = "#ffad60ff"
  ), labels = c(
    "s_Harris_NSC_IPC" = "Whole gene set",
    "s_Top10_NSCIPC_Harris" = "Top 10 genes", 
    "s_Top40_NSCIPC_Harris" = "Top 40 genes", 
    "s_Overlap_NP" = "Shared NP genes Fig.1"
  ), name = NULL) +
  xlim(0, 1) +
  ylim(-2, 2) +
  theme_tufte(base_family = "Arial") +
  theme(aspect.ratio = 0.8)  # Adjust aspect ratio to make it narrower

# Display plot
print(s_combined_plot_NSCIPC_Harris)



```

### 4. Figure 3

Panel A was created by using Go term lists provided by both respective studies and summarized. 
Panel B was created using code produced by Harris et al. and can be found [here](https://github.com/harrislachlan/lifelong_stemcells) to aquire the seurat object and nsc.integrated, 
module score of the SenMayo list was added with the [SenMayo genelist](https://www.nature.com/articles/s41467-022-32552-1) and the package [UCell](https://bioconductor.org/packages/release/bioc/html/UCell.html): 

```{r Code senescence, eval=FALSE }
#Prerequisite: nsc.integrated and seurat object from Harris et al. 
# Load the senescence gene lists
senMayo <- read.csv("SenMayo.csv", header = FALSE)

# Convert data frames to vectors
senMayo <- as.vector(senMayo$V1)

# Define the signatures list
signatures <- list(
  SenMayoList = senMayo
)

# Calculate module scores using AddModuleScore_UCell
nsc.integrated <- AddModuleScore_UCell(nsc.integrated, features = signatures, name = "Senescence_Score")

seurat.object <- AddModuleScore_UCell(nsc.integrated, features = signatures, name = NULL)

#Visualize module scores using FeaturePlot
FeaturePlot(seurat.object, pt.size = 1, min.cutoff = 0.04 , reduction = "umap", features = names(signatures))
```

### 7. Session info

Shown below are all all packages and versions necessary to reproduce the results in this report.

```{r session info}
sessionInfo()
```
